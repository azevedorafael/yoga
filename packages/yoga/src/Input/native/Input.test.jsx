import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { ThemeProvider, Input } from '../..';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should match with default input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with label', () => {
      const { container } = render(
        <ThemeProvider>
          <Input label="Input" />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match when input is focused', () => {
      const { container, getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(container).toMatchSnapshot();
    });

    it('should match with disabled input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input disabled />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with error', () => {
      const { container } = render(
        <ThemeProvider>
          <Input error="Error message" />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with helper text and max length', () => {
      const { container } = render(
        <ThemeProvider>
          <Input helper="Helper text" maxLength={20} />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with full width', () => {
      const { container } = render(
        <ThemeProvider>
          <Input label="Label" full />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChangeText', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" onChangeText={onChangeTextMock} />
        </ThemeProvider>,
      );

      fireEvent.changeText(getByTestId('input'), 'foo');

      expect(onChangeTextMock).toHaveBeenCalled();
    });

    it('should not call onChangeText when input is disabled', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input
            label="Input"
            testID="input"
            onChangeText={onChangeTextMock}
            disabled
          />
        </ThemeProvider>,
      );

      fireEvent.changeText(getByTestId('input'), 'foo');

      expect(onChangeTextMock).not.toHaveBeenCalled();
    });

    it('should call onFocus', () => {
      const onFocusMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" onFocus={onFocusMock} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(onFocusMock).toHaveBeenCalled();
    });

    it('should call onBlur', () => {
      const onBlurMock = jest.fn();

      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" onBlur={onBlurMock} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));
      fireEvent.blur(getByTestId('input'));

      expect(onBlurMock).toHaveBeenCalled();
    });
  });

  describe('maxLength', () => {
    it('should update maxLength counter when add character', () => {
      const { getByTestId, getByText } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" maxLength={10} />
        </ThemeProvider>,
      );

      expect(getByText('0/10').children.join('')).toBe('0/10');

      fireEvent.changeText(getByTestId('input'), 'foo');

      expect(getByText('3/10').children.join('')).toBe('3/10');
    });
  });

  describe('clean button', () => {
    it('should clear input value when press clean button', () => {
      const { getByRole, getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" />
        </ThemeProvider>,
      );

      fireEvent.changeText(getByTestId('input'), 'foo');
      fireEvent.press(getByRole('button'));

      expect(getByTestId('input').props.value).toBe('');
    });
  });
});
