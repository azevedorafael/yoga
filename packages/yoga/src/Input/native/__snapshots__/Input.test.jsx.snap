// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match when input is focused 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={false}
      editable={true}
      focus={true}
      full={false}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#41414A",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#41414A",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": 320,
          },
          Object {
            "height": 52,
          },
        ]
      }
      testID="input"
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      focus={true}
      pointerEvents="none"
      style={
        Object {
          "left": 14,
          "position": "absolute",
          "top": 24,
          "transform": Array [
            Object {
              "translateY": -8,
            },
          ],
        }
      }
      typed={false}
    >
      <Text
        disabled={false}
        focus={true}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#41414A",
            "fontSize": 14,
            "fontWeight": "700",
            "paddingLeft": 4,
            "paddingRight": 4,
          }
        }
        typed={false}
      >
        Input
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with default input 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={false}
      editable={true}
      focus={false}
      full={false}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": 320,
          },
          Object {
            "height": 52,
          },
        ]
      }
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with disabled input 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={true}
      editable={false}
      focus={false}
      full={false}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#D3D3E2",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": 320,
          },
          Object {
            "height": 52,
          },
        ]
      }
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with error 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={false}
      editable={true}
      error="Error message"
      focus={false}
      full={false}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#CB3530",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": 320,
          },
          Object {
            "height": 52,
          },
        ]
      }
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      full={false}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 4,
            "maxWidth": 320,
            "width": 320,
          },
        ]
      }
    >
      <Text
        disabled={false}
        error="Error message"
        style={
          Array [
            Object {
              "color": "#CB3530",
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontSize": 12,
            },
          ]
        }
      >
        Error message
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with full width 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={true}
    style={
      Array [
        Object {
          "position": "relative",
          "width": "100%",
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={false}
      editable={true}
      focus={false}
      full={true}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": "100%",
          },
          Object {
            "height": 52,
          },
        ]
      }
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      focus={false}
      pointerEvents="none"
      style={
        Object {
          "left": 16,
          "position": "absolute",
          "top": 24,
          "transform": Array [
            Object {
              "translateY": -8,
            },
          ],
        }
      }
      typed={false}
    >
      <Text
        disabled={false}
        focus={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
        typed={false}
      >
        Label
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with helper text and max length 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={false}
      editable={true}
      focus={false}
      full={false}
      maxLength={20}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": 320,
          },
          Object {
            "height": 52,
          },
        ]
      }
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      full={false}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 4,
            "maxWidth": 320,
            "width": 320,
          },
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#9898A7",
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontSize": 12,
            },
          ]
        }
      >
        Helper text
      </Text>
      <Text
        disabled={false}
        right={true}
        style={
          Array [
            Object {
              "color": "#9898A7",
              "flexShrink": 0,
              "flexWrap": "wrap",
              "fontSize": 12,
              "marginLeft": "auto",
            },
          ]
        }
      >
        0
        /
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with label 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      cleanable={true}
      disabled={false}
      editable={true}
      focus={false}
      full={false}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 40,
            "paddingTop": 16,
            "width": 320,
          },
          Object {
            "height": 52,
          },
        ]
      }
      typed={false}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      focus={false}
      pointerEvents="none"
      style={
        Object {
          "left": 16,
          "position": "absolute",
          "top": 24,
          "transform": Array [
            Object {
              "translateY": -8,
            },
          ],
        }
      }
      typed={false}
    >
      <Text
        disabled={false}
        focus={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#9898A7",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
        typed={false}
      >
        Input
      </Text>
    </View>
  </View>
</View>
`;
